pkg/cmd/server/start/master_args.go

```diff
@@ -193,6 +189,14 @@ func (args MasterArgs) BuildSerializeableMasterConfig() (*configapi.MasterConfig
 		},
 	}
 
+	if args.BindAddrArg.BindAddr.URL.Scheme == "https" {
```

Comment:
I think this check should be based on the presence of absence of certs.  That way the bind address can remain a host:port and not require different formats depending on transport type.